README.md

```diff
@@ -5,757 +5,758 @@
-and transitions for items in the array.
-
-```js
-let today = create(TodoApp, { todos: [{ title: 'Buy milk' }] }).state;
-// => { todos: [ Todo { title: 'Buy milk', isCompleted: false } ] }
-``` 
+## `set` transition
 
-It will also provide you with transitions array with objects for each data item. You can invoke,
-transitions from this array and they will behave as all other transitions in microstates.
+The `set` transition is the only transition that is available on all types. It can be used to replace the value of the current Microstate with another value.
 
 ```js
-today.todos[0].title.set('Buy 2% milk').state;
-// => { todos: [ Todo { title: 'Buy 2% milk', isCompleted: false } ] }
+import { create } from 'microstates'
 
-today.todos.push({ title: 'Review PRs' }).state
-// => { todos: [ Todo { title: 'Buy 2% milk', isCompleted: false }, Todo { title: 'Review PRs', isCompleted: false } ] }
+let number = create(Number, 42).set(43);
+
+number.valueOf()
+//> 43
 ```
 
-You can do the same with Objects using `{Type}` syntax. 
+You can also use `set` transition to replace the current Microstate with another Microstate. This is especially useful when building state machines because it allows you to change the type of the current Microstate. By changing the type, you're also changing available transitions and how the state is calculated.
 
 ```js
-let today = create({Todo}, { t1: { title: 'Buy milk' }, t2: { title: 'Review PRs' } }).state
-// => { t1: Todo { title: 'Buy milk', isComplete: false }, t2: Todo { title: 'Review PRs', isComplete: false } }
-
-today.t2.isComplete.toggle().state
-// => { t1: Todo { title: 'Buy milk', isComplete: false }, t2: Todo { title: 'Review PRs', isComplete: true } }
-```
+import { types } from 'microstates';
 
-## Observable interoperability
+class Vehicle {
+  weight = Number;
+  towing = Vehicle;
 
-You can create an observable from any microstate using the `Observable.from` method.
-The resulting Observable will stream the next microstate for every transition.
-When you subscribe to the observable microstate, you'll immediately receive a microstate through the stream.
-This microstate will have transitions that you can call to cause the next microstate to come through the stream.
+  get isTowing {
+    return !!this.towing;
+  }
 
-```js
-import { create } from 'microstates';
-import { Observable } from 'rxjs';
+  get towCapacity() {
+    throw new Error('not implemented');
+  }
 
-let ms = create(Number, 42);
-let observable = Observable.from(ms);
+  tow() {
+    throw new Erorr('not implemented');
+  }
+}
 
-let last;
+class Car extends Vehicle {}
 
-observable.subscribe(microstate => {
-  last = microstate;
-  console.log(microstate.state);
-});
-// => 42
+class Truck extends Vehicle {
+  towCapacity = Number;
 
-last.increment().increment().increment();
-// => 43
-// => 44
-// => 45
-```
+  tow(vehicle) {
+    if (vehicle.weight.state < this.towCapacity.state) {
+      return this.vehicle.set(vehicle);
```

Comment:
I think this is meant to be `this.towing.set(vehicle)` because the `Vehicle` class defines it as `towing`.

Reply:
Yes, you're right. Thank you. Good catch.